src/main/kotlin/com/cjbooms/fabrikt/generators/controller/ControllerGeneratorUtils.kt

```diff
@@ -48,4 +49,41 @@ object ControllerGeneratorUtils {
 
     private fun httpVerbMethodName(verb: String, isSingleResource: Boolean) =
         if (isSingleResource) "${verb}ById" else verb
+
+
+    /**
+     * Enum definition for different cases of authentication
+     *
+     * N0_SECURITY when no @Secured Annotation is to be added neither an authentication input parameter
+     * AUTHENTICATION_REQUIRED when @Secured Annotation IS_AUTHENTICATED (for micronaut) is to be added and authentication as input parameter
+     * AUTHENTICATION_PROHIBITED when @Secured Annotation IS_ANONYMOUS (for micronaut) is to be added and no authentication input parameter
+     * AUTHENTICATION_OPTIONAL when @Secured Annotation IS_AUTHENTICATED and IS_ANONYMOUS (for micronaut) is to be added and authentication as optional input parameter
+     */
+    enum class SecuritySupport(val allowsAuthenticated: Boolean, val allowsAnonymous: Boolean) {
+        NO_SECURITY(false, false),
```

Comment:
**nitpick:** The canonical way to document enum values is to do so for each value:
```kotlin
        /**
         * When the operation does not support any way of security checks.
         */
        NO_SECURITY(false, false),
```

Reply:
I see.